cabinets/cabinet/s3_cabinet.py

```diff
@@ -18,6 +18,9 @@ def set_configuration(cls, region_name='us-east-1', aws_access_key_id=None,
 
     @classmethod
     def read_content(cls, path, **kwargs) -> bytes:
+        if not cls.client:
+            cls.client = boto3.client('s3')
```

Comment:
Having to put this in many places make me wonder if it's a good pattern. I feel like we should be able to guarantee a client exists before ever accessing these methods.

Reply:
What about instantiating it on initialization?

Reply:
I know this is because of `moto`, but at minimum maybe we could at least make a helper method like `_ensure_client_exists()`. Feels really odd though. Starting to wonder if relying on class methods like this is just bad design.